daemon.go

```diff
@@ -22,58 +21,41 @@ import (
 
 // Config section
 
-//TODO: daemon configuration will be hardcoded now
+//TODO: some of daemon configuration will be hardcoded now
 const (
-	// poll data from server every 3 minutes by default
-	defaultServerpollInterval = time.Duration(3) * time.Minute
-	defaultServerAddress      = "menderserver"
-	defaultDeviceID           = "ABCD-12345"
-	defaultAPIversion         = "0.0.1"
+	defaultDeviceID   = "ABCD-12345"
+	defaultAPIversion = "0.0.1"
 )
 
 // daemon configuration
-type daemonConfigType struct {
+type daemonConfig struct {
 	serverpollInterval time.Duration
-	server             string
+	serverURL          string
 	deviceID           string
 }
 
-func (daemon *menderDaemon) LoadConfig(configFile string) error {
-	//TODO: change to properly load config from file
-	var config daemonConfigType
-	config.serverpollInterval = defaultServerpollInterval
-	config.server = getMenderServer(configFile)
-	config.deviceID = defaultDeviceID
-
-	daemon.config = config
-	return nil
-}
-
-func getMenderServer(serverFile string) string {
-	// TODO: this should be taken from configuration or should be set at bootstrap
-	server, err := ioutil.ReadFile(serverFile)
-
-	log.Debug("Reading Mender server name from file " + serverFile)
-
-	// return default server address if we can not read it from file
-	if err != nil {
-		log.Warn("Can not read server file " + err.Error())
-		return defaultServerAddress
-	}
-	return string(server)
-}
-
 // Daemon section
 
 type menderDaemon struct {
 	Updater
 	UInstallCommitRebooter
-	config      daemonConfigType
+	Controler
+	config      daemonConfig
 	stopChannel chan (bool)
 }
 
-func NewDaemon(client Updater, device UInstallCommitRebooter) *menderDaemon {
-	daemon := menderDaemon{client, device, daemonConfigType{}, make(chan bool)}
+func NewDaemon(client Updater, device UInstallCommitRebooter,
+	mender Controler) *menderDaemon {
+
+	config := mender.GetDaemonConfig()
+
+	daemon := menderDaemon{
+		client,
```

Comment:
This is very much neat picking:
When initialising structure like this it's nice to use named params it's more readable and more resilient to changes to the original structure fields.